### PR TITLE
CIV-14334 Submit function for Hwf - GA for Lips

### PR DESCRIPTION
**Pull request by deepthidoppalapudihmcts**
**Before creating a pull request make sure that:**

- [x ] commit messages are meaningful and follow good commit message guidelines
- [x ] README and other documentation has been updated / added (if needed)
- [x ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-14334

### Change description ###

Work Allocation tests for HelpWithFees Application fee and Additional fee

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -8,10 +8,10 @@ def type = "nodejs"
 def product = "civil"
 def component = "citizen-ui"
 def camundaBranch = "master"
-def dmnBranch = "master"
+def dmnBranch = "CIV-14334"
 def waStandaloneBranch = "master"
 def ccdBranch = "master"
-def generalappCCDBranch = "master"
+def generalappCCDBranch = "CIV-14334"
```
Comment (neeta-hmcts) [inline Jenkinsfile_CNP:14]: make sure to remove this once you complete the testing on the preview.

File: src/main/common/models/gaEvents/eventDto.ts [hunk 2]
```diff
@@ -20,20 +20,19 @@ import {CaseLink} from 'models/generalApplication/CaseLink';
...
-  generalAppRespondentAgreement?: CcdGeneralApplicationRespondentAgreement;
-  generalAppInformOtherParty?: CcdGeneralApplicationInformOtherParty;
-  generalAppAskForCosts?: YesNoUpperCamelCase;
-  generalAppDetailsOfOrder?: string;
-  generalAppReasonsOfOrder?: string;
-  generalAppEvidenceDocument?: CcdGeneralApplicationEvidenceDocument[];
-  generalAppHearingDetails?: CcdGeneralApplicationHearingDetails;
-  generalAppStatementOfTruth?: CcdGeneralApplicationStatementOfTruth;
-  generalAppHelpWithFees?: CCDHelpWithFees;
+  generalAppType: CcdGeneralApplicationTypes;
```
Comment (neeta-hmcts) [inline src/main/common/models/gaEvents/eventDto.ts:27]: as you have made all the fields mandatory. please make sure the '**Upload Additional Documents**' event is working fine.

File: src/main/routes/features/generalApplication/applicationFee/helpWithApplicationFeeReferenceController.ts [hunk 3]
```diff
@@ -35,43 +39,50 @@ async function renderView(res: Response, req: AppRequest | Request, form: Generi
...
       const redirectUrl = constructResponseUrlWithIdParams(claimId, GA_APPLY_HELP_WITH_FEE_REFERENCE);
-      await renderView(res, req, form, claimId, redirectUrl);
+      await renderView(res, req, form, claimId, redirectUrl, false);
     } else {
       const redisKey = generateRedisKey(<AppRequest>req);
       await saveHelpWithFeesDetails(redisKey, new ApplyHelpFeesReferenceForm(req.body.option, req.body.referenceNumber), hwfPropertyName);
-      res.redirect(getRedirectUrl(claimId, form.model));
+
+      if (form.model.option === YesNo.YES) {
+        await saveAndTriggerNotifyGaHwfEvent(claimId, <AppRequest>req, form.model);
```
Comment (neeta-hmcts) [inline src/main/routes/features/generalApplication/applicationFee/helpWithApplicationFeeReferenceController.ts:73]: typecast not req

File: Jenkinsfile_CNP (hunk 1 above)
Comment (deepthidoppalapudihmcts) [reply Jenkinsfile_CNP:14]: i will remove this after testing

File: src/main/common/models/gaEvents/eventDto.ts (hunk 2 above)
Comment (deepthidoppalapudihmcts) [reply src/main/common/models/gaEvents/eventDto.ts:27]: Its a merge conflict issue, will clear back that

File: src/main/routes/features/generalApplication/applicationFee/helpWithApplicationFeeReferenceController.ts (hunk 3 above)
Comment (deepthidoppalapudihmcts) [reply src/main/routes/features/generalApplication/applicationFee/helpWithApplicationFeeReferenceController.ts:73]: Argument of type 'AppRequest| Request is not assignable to parameter of type 'AppRequest
If not typecasted

**Pull request merged**